yang/frr-staticd.yang

```diff
@@ -0,0 +1,141 @@
+module frr-staticd {
+  yang-version "1.1";
+  namespace "http://frrouting.org/yang/staticd";
+
+  prefix frr-staticd;
+
+  import frr-routing {
+    prefix "frr-rt";
+  }
+
+  import frr-nexthop {
+    prefix frr-nexthop;
+  }
+
+  import ietf-inet-types {
+    prefix inet;
+  }
+
+  organization
+    "Free Range Routing";
+
+  contact
+    "FRR Users List: <mailto:frog@lists.frrouting.org>
+     FRR Development List: <mailto:dev@lists.frrouting.org>";
+
+  description
+    "This module defines a model for managing FRR staticd information.
+     This YANG module augments the ietf-routing with additional
+     nexthop information";
+
+  revision 2019-12-03 {
+    description
+      "Initial revision.";
+  }
+
+
+  augment "/frr-rt:routing/frr-rt:control-plane-protocols/"
+        + "frr-rt:control-plane-protocol" {
+    container staticd {
+      when "../frr-rt:type = 'frr-rt:static'" {
+        description
+          "This container is only valid for the 'static' routing
+           protocol.";
+      }
+      description
+      "Support for a 'static' pseudo-protocol instance
+       consists of a list of routes.";
+
+      list address-family {
```

Comment:
Where does vrf come into this? I only see unqualified ip prefixes: are the vrfs captured somewhere else?

Reply:
vrf would come from the frr-routing.yang
It would be like this
module: frr-routing
  +--rw routing
     +--rw control-plane-protocols
        +--rw control-plane-protocol* [type name vrf]
           +--rw type                   identityref
           +--rw name                   string
           +--rw vrf                    frr-vrf:vrf-ref
           +--rw frr-staticd:staticd
              +--rw frr-staticd:address-family* [address-family]